File: training-data/kotlin/LookAndSay.kt
fun nextTerm(s: String): String = buildString {
    var i = 0
    while (i < s.length) {
        var j = i
        while (j < s.length && s[j] == s[i]) j++
        append(j - i).append(s[i])
        i = j
    }
}

fun main() {
    generateSequence("1", ::nextTerm).take(8).forEach(::println)
}

File: training-data/dart/look_and_say.dart
String nextTerm(String s) => s.replaceAllMapped(RegExp(r'(\d)\1*'), (m) => '${m[0]!.length}${m[1]}');

Iterable<String> lookAndSay() sync* {
  var s = '1';
  while (true) {
    yield s;
    s = nextTerm(s);
  }
}

void main() => lookAndSay().take(8).forEach(print);
